File: backend/src/main/java/com/pratik/hotelreservation/service/RoomService.java
package com.pratik.hotelreservation.service;

import com.pratik.hotelreservation.dto.request.RoomCreateRequest;
import com.pratik.hotelreservation.dto.request.RoomUpdateRequest;
import com.pratik.hotelreservation.dto.response.RoomResponse;

import java.util.List;

public interface RoomService {

    RoomResponse createRoom(RoomCreateRequest request);

    RoomResponse getRoomById(Long id);

    List<RoomResponse> getAllRooms();

    List<RoomResponse> getRoomsByHotel(Long hotelId);

    RoomResponse updateRoom(Long id, RoomUpdateRequest request);

    void deleteRoom(Long id);
}

File: frontend/src/services/roomService.js
import api from "./api";

export const getAllRooms = async () => {
  const response = await api.get("/rooms");
  return response.data.data;
};

export const getRoomById = async (roomId) => {
  const response = await api.get(`/rooms/${roomId}`);
  return response.data.data;
};

export const getRoomsByHotel = async (hotelId) => {
  const response = await api.get(`/rooms/hotel/${hotelId}`);
  return response.data.data;
};